### ISSUE TITLE
Drupal 10 and CKEditor 5 issues/differences

### ISSUE THREAD

**Issue by keelanfh** [localgovdrupal/localgov_core#181]
Thought I'd do a bit of testing with CKEditor in our D10 branch.

A few things I observed:

## Height of body field

The height of the body field (in Service pages, perhaps elsewhere too) is very small compared to in CKEditor 4. Actually smaller than the Summary field.

![image](https://github.com/localgovdrupal/localgov_core/assets/19519457/311e14a2-c611-4497-af75-71c2ca6bacc0)

### D9 for comparison:
![image](https://github.com/localgovdrupal/localgov_core/assets/19519457/169b9b74-eb5c-4074-b41b-9ff41dc34b44)

## Possible to e.g. set link styles on non-links

In D9 you cannot set link styles (external link, PDF link, Start button) on non-links. Similarly for some of the other styles ("Green ticks") which should only be set on certain elements. In D10 you can do this, which means you have something styled as an interactive element which actually isn't interactive.

## Possible to set multiple styles on the same element

In D10 it is possible to set multiple conflicting styles on the same element, e.g. here is an item styled simultaneously as Callout primary, Callout teal and Callout carbon. It should only be possible to set one of these styles

![image](https://github.com/localgovdrupal/localgov_core/assets/19519457/66280e26-bcb9-4a5d-8e18-c262ebd9f4ec)

## Styles do not appear in WYSIWYG preview

In D9 there are some nice preview styles in the WYSIWYG to show what things will look like when you publish them. These are missing in D10:

![image](https://github.com/localgovdrupal/localgov_core/assets/19519457/89b0891c-ae9c-4822-acf4-3178250a44ed)

D9 for comparison:

![image](https://github.com/localgovdrupal/localgov_core/assets/19519457/e0f16187-e900-4a84-87ab-f4706307ce3e)

## New "code" and "code block" styles have appeared

Unlikely to be useful for LGD, should probably be removed

![image](https://github.com/localgovdrupal/localgov_core/assets/19519457/83572bde-4f3e-45d9-a7db-ae9093e3f220)

**Comment by ekes** [comment]
Is that https://www.drupal.org/project/drupal/issues/3241295 fixed in 10.1.x or is the min max rows setting not there?

**Comment by keelanfh** [comment]
> Is that https://www.drupal.org/project/drupal/issues/3241295 fixed in 10.1.x or is the min max rows setting not there?

I'll check. Looks like I was on 10.0.9.

**Comment by justinepocock** [comment]
The height of the body field can be fixed with css much easier in CK5 I've added to a ckeditor.css file
```
.ck-editor__editable {
  min-height: 200px;
} 
```

Styles do not appear in WYSIWYG preview - you need to make a stylesheet for the styles to appear and add the following code to the themes info.yml file.  It might be as simple as adding the 5 to the line of code
```
ckeditor5-stylesheets: 
  - css/base/ckeditor.css 
```

I'll see if I can get some patches whipped up over the next few days for those, I'm following the chat in the Drupal #ckeditor5 slack channel closely, and work is progressing fast but not fast enough, especially around the setting styles on elements.

**Comment by keelanfh** [comment]
Awesome!

**Comment by keelanfh** [comment]
For some progress on this one:

- [x] Height of body field - fixed in https://github.com/localgovdrupal/localgov_base/pull/462
- [x] Possible to set link styles on non-links - fixed upstream and released in Drupal 10.1.0
- [ ] Possible to set multiple styles on the same element - not fixed. I think this was an upstream design decision to allow multiple styles to be applied to the same element.
- [x] Styles do not appear in WYSIWYG preview - fixed in https://github.com/localgovdrupal/localgov_base/pull/462
- [ ] New "code" and "code block" styles have appeared - fix in https://github.com/localgovdrupal/localgov_core/pull/185

**Comment by keelanfh** [comment]
Closing this as I think most things got sorted in the end and not much value in keeping this issue open.
